romanisim/tests/test_image.py

```diff
@@ -746,3 +746,113 @@ def test_inject_source_into_image():
                    'flux': flux,
                    'tij': tij}
         af.write_to(os.path.join(artifactdir, 'dms231.asdf'))
+
+
+def test_inject_source_into_mosaic():
+    """Inject a source into a mosaic.
+    """
+
+    # Set constants and metadata
+    galsim.roman.n_pix = 100
+    rng_seed = 42
+    rng = galsim.UniformDeviate(rng_seed)
+    metadata = copy.deepcopy(parameters.default_parameters_dictionary)
+    metadata['instrument']['detector'] = 'WFI07'
+    metadata['instrument']['optical_element'] = 'F158'
+    metadata['exposure']['ma_table_number'] = 1
+    bandpass = [metadata['instrument']['optical_element']]
+    filter_name = metadata['instrument']['optical_element']
+    sca = 4
+
+    # Create PSF
+    twcs = wcs.get_wcs(metadata, usecrds=False)
+    psf = romanisim.psf.make_psf(sca, filter_name, wcs=twcs, chromatic=False, webbpsf=True)
+
+    # Create initial Level 2-like image
+
+    # Create Four-quadrant pattern of gaussian noise, centered around one
+    # Each quadrant's gaussin noise scales like total exposure time
+    # (total files contributed to each quadrant)
+
+    # Create gaussian noise generators
+    g1 = galsim.GaussianDeviate(31415926, mean=1.0, sigma=0.01)
+    g2 = galsim.GaussianDeviate(31415926, mean=1.0, sigma=0.02)
+    g3 = galsim.GaussianDeviate(31415926, mean=1.0, sigma=0.05)
+    g4 = galsim.GaussianDeviate(31415926, mean=1.0, sigma=0.1)
+
+    # Create cps Image data array
+    la_arr = np.zeros([200, 200])
+
+    # Populate the image array with gaussian noise from generators
+    g1.generate(la_arr[0:100, 0:100])
+    g2.generate(la_arr[0:100, 100:200])
+    g3.generate(la_arr[100:200, 0:100])
+    g4.generate(la_arr[100:200, 100:200])
+
+    # Poisson Noise of image array
+    var_poisson = (la_arr - 1)**2
```

Comment:
I would make this image piecewise constant nad equal to 0.01^2, 0.02^2, 0.05^2, 0.1^2, and not have the "noise" component of the uncertainty, if that makes sense.  This is just for simplicity, really; in any case, we're just setting up an example image in this phase.